src/iso.rs

```diff
@@ -733,8 +735,12 @@ fn iso_date_surpasses(this: &IsoDate, other: &IsoDate, sign: i8) -> bool {
 
 #[inline]
 fn balance_iso_year_month(year: i32, month: i32) -> (i32, i32) {
-    let y = year + (month - 1) / 12;
+    // 1. Assert: year and month are integers.
+    // 2. Set year to year + floor((month - 1) / 12).
+    let y = year + (month - 1).div_euclid(12);
```

Comment:
Oof, that's a really tricky bug.

Reply:
It took longer than I'd like to admit to notice it too 😭